core/src/main/java/org/elasticsearch/common/util/concurrent/KeyedLock.java

```diff
@@ -115,58 +112,4 @@ public boolean hasLockedKeys() {
         return !map.isEmpty();
     }
 
-    /**
-     * A {@link KeyedLock} that allows to acquire a global lock that guarantees
-     * exclusive access to the resource the KeyedLock is guarding.
-     */
-    public final static class GlobalLockable<T> extends KeyedLock<T> {
```

Comment:
I could have removed it in my previous commit. We used it to protect the cache of fielddata instances, but this cache has been removed, so we don't need it anymore.